### ISSUE TITLE
Feature request: add a way to reduce stdout output from meson build 

### ISSUE THREAD

**Issue by danielcjacobs** [mesonbuild/meson#10992]
It would be nice to have more control over what gets printed to stdout when running `meson build`. Most of the information I usually get is just noise that could be written only to `meson-log.txt`, but I can see why it _can_ be useful to see in stdout.

For example, here's the configuration output for the `fmt` subproject contained in one of my projects:
```
Run-time dependency fmt found: NO (tried pkgconfig and cmake)
Looking for a fallback subproject for the dependency fmt

Executing subproject fmt 

fmt| Project name: fmt
fmt| Project version: 7.1.3
fmt| C++ compiler for the host machine: ccache c++ (gcc 12.2.1 "c++ (GCC) 12.2.1 20220819 (Red Hat 12.2.1-2)")
fmt| C++ linker for the host machine: c++ ld.bfd 2.37-36
fmt| Build targets in project: 36
fmt| Subproject fmt finished.

Dependency fmt from subproject subprojects/fmt-7.1.3 found: YES 7.1.3
```

To me, most of this information is `debug` information. All I really care to see is:
```
Run-time dependency fmt found: NO (tried pkgconfig and cmake)
Looking for a fallback subproject for the dependency fmt

Executing subproject fmt 
Dependency fmt from subproject subprojects/fmt-7.1.3 found: YES 7.1.3
```

I might even be able to get by with less than that, since I know it will all show up in `meson-log.txt` anyway.

This is somewhat of a difficult thing to resolve, as different users might want more or less info. I would argue that erring on the side of less is much better, since anyone who wants more can still find it in the logfile. Plus, most of the data in the output I pasted is objectively _debug_ information (e.g. the C++ linker).

Suggestions (not mutually-exclusive):

#### Option 1: Make the data logged to stdout configurable via meson arg

Ideally, the data that is logged to stdout would be configurable, but this might be an unreasonable expectation. I envision a configuration parameter that lets you set a level for the meson output, and that more or less messages would be converted from `mlog.log` to `mlog.debug` based on the level the user sets.

#### Option 2: Make the data logged to stdout configurable via function parameter

 The log level could also be a parameter that can be passed to different meson functions like `subproject()` or `dependency()`.  In the example output I pasted above, all of the log messages from inside the `fmt` subproject could be ignored. Perhaps at least there could be an option to quiet subproject log messages.

#### Option 3: Convert some `mlog.log()` calls to `mlog.debug()`

If the above requests are unlikely to be approved (I think at least option 2 is a reasonable one), I may try and open an MR that just changes quite a few `mlog.log` messages to `mlog.debug`.

**Comment by eli-schwartz** [comment]
Option 4: make some `mlog.log()` be `mlog.debug()` but only inside a subproject.

**Comment by danielcjacobs** [comment]
That would eliminate about 90% of what I consider to be noise (depending on how much are converted to `mlog.debug()`), so I approve of that option.

**Comment by LunarLambda** [comment]
In particular, compilers, linkers, etc. are definitely global right? There's no reason to print them for every single subproject. I have a toolchain project which bundles > a dozen libraries as subprojects, and in a dev build, all of them are built, causing `meson setup` to create an unreadable flood of output - most of which is just the same compiler being printed 15 times.

Similarly, `find_program` and `dependency` print every single time they're used, even if they return the same thing. For example:

```meson
dependency('zlib')
dependency('zlib')
dependency('zlib')
```

definitely should return the same dependency. But the "found" message is printed 3 times, for no real reason. It'd be nice if these things could be cached so that only the first hit prints a message.

I know this can be solved by saving the program/dependency in a variable, but keeping track of dependency variables particularly across subprojects/subdirs is finicky, it's much more straightforward to just write the `dependency`s you need in a subdir explicitly again (common with tests or examples, where the same dependency is used for many separate targets).

There's also some redundancy in subproject logs in general, like `name| Project name: name` and `name| Subproject name finished`. I understand this is because subprojects reuse a lot of the code paths of the main project, but it really does create a ton of extraneous output. I also believe that currently, every subproject re-runs the compiler sanity checks, even though, again, compilers are global (and there are no plans for them to stop being global?), so there are additional inefficiencies in `meson setup`, particularly on slower machines or if re-running setup frequently (such as changing cross/native files)

All this makes the output very hard to parse by a human, which somewhat defeats the point of even having output. Most times, if there's no outright error, I just assume everything is A-OK, because the important information is too buried.

**Comment by zmitchell** [comment]
It would be really nice to have a `--quiet` flag for most if not all subcommands
